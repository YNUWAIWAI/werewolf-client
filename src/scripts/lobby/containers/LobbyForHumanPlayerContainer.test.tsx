import * as ActionTypes from '../constants/ActionTypes'
import * as React from 'react'
import {IntlProviderContainer} from './IntlProviderContainer'
import {LobbyForHumanPlayer} from '../components/templates/LobbyForHumanPlayer'
import {LobbyForHumanPlayerContainer} from './LobbyForHumanPlayerContainer'
import {Provider} from 'react-redux'
import fakeStore from './fakeStore'
import {mount} from 'enzyme'

describe('<LobbyForHumanPlayerContainer />', () => {
  test('render', () => {
    const store = fakeStore()
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <LobbyForHumanPlayerContainer />
        </IntlProviderContainer>
      </Provider>
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
  test('selectVillage', () => {
    const store = fakeStore()
    const dispatch = jest.fn()

    store.dispatch = dispatch
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <LobbyForHumanPlayerContainer />
        </IntlProviderContainer>
      </Provider>
    )
    const id = 1

    wrapper.find(LobbyForHumanPlayer).props().selectVillage(id)()
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch).toHaveBeenCalledWith({
      id,
      type: ActionTypes.App.SELECT_VILLAGE
    })
  })
})
