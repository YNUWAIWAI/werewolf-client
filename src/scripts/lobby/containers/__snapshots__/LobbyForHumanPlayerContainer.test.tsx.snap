// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LobbyForHumanPlayerContainer /> render 1`] = `"<div class=\\"lo--grid\\"><div class=\\"lo--header\\">Lobby for human players</div><div class=\\"lo--avatar\\"><img class=\\"lo--avatar--image\\" src=\\"\\"><span class=\\"lo--avatar--name\\"></span></div><div class=\\"lo--main-content\\"><div class=\\"lo--village\\"></div></div><div class=\\"lo--aside-content \\"><ul class=\\"lo--compact-menu\\"><li class=\\"lo--compact-menu--item  \\">Build your village</li><li class=\\"lo--compact-menu--item  \\">ID search</li><li class=\\"lo--compact-menu--item  \\">Advanced search</li><li class=\\"lo--compact-menu--item  \\">Refresh</li><li class=\\"lo--compact-menu--item  \\">Return to the main page</li></ul></div></div>"`;
